### PR TITLE
Add variables in controller for 15sp2 and 15sp3 client, minion and ss…

### PR DESCRIPTION
**Pull request by ktsamis**
## What does this PR change?

Add variables in controller for 15sp2 and 15sp3 client, minion and sshminion

Fixes https://github.com/SUSE/spacewalk/issues/13581, maybe not completely, have to run a deployment.

### CODE REVIEW

File: salt/mirror/mirror.sh.conf [hunk 1]
```diff
@@ -4,6 +4,8 @@ https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/
 https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse151.x86_64.qcow2
 http://download.suse.de/ibs/Devel:/Galaxy:/Terraform:/Images/images/sles15.x86_64.qcow2
 http://download.suse.de/ibs/Devel:/Galaxy:/Terraform:/Images/images/sles15sp1.x86_64.qcow2
+http://download.suse.de/ibs/Devel:/Galaxy:/Terraform:/Images/images/sles15sp2.x86_64.qcow2
+http://download.suse.de/ibs/Devel:/JeOS:/SLE-15-SP3/images/SLES15-SP3-JeOS.x86_64-kvm-and-xen.qcow2
```
Comment (ktsamis) [inline salt/mirror/mirror.sh.conf:8]: For this I'm not sure if we should use it for long, but not sure who can copy the image in the right folder.

Comment (juliogonzalez) [reply salt/mirror/mirror.sh.conf:8]: What do you mean?

BTW: What's this image? Is the latest milestone released? If so, we could use it at the tfstates as well! :-)

File: salt/mirror/mirror.sh.conf [hunk 2]
```diff
@@ -4,6 +4,8 @@ https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/
 https://download.opensuse.org/repositories/systemsmanagement:/sumaform:/images:/libvirt/images/opensuse151.x86_64.qcow2
 http://download.suse.de/ibs/Devel:/Galaxy:/Terraform:/Images/images/sles15.x86_64.qcow2
 http://download.suse.de/ibs/Devel:/Galaxy:/Terraform:/Images/images/sles15sp1.x86_64.qcow2
+http://download.suse.de/ibs/Devel:/Galaxy:/Terraform:/Images/images/sles15sp2.x86_64.qcow2
```
Comment (juliogonzalez) [inline salt/mirror/mirror.sh.conf:7]: I am confused by this, aren't we supposed to use official QCOW images for SLE15, openSUSE, etc? @moio?

Should we maybe update mirror.sh.conf?

Comment (srbarrios) [reply salt/mirror/mirror.sh.conf:7]: Agree with Julio, we shouldn't use those custom images here. I guess we don't use the mirror module that often, so people didn't notice. But this doesn't match the same file on our internal infrastructure salt configuration for our internal mirrors.

File: salt/mirror/mirror.sh.conf (hunk 1 above)
Comment (srbarrios) [reply salt/mirror/mirror.sh.conf:8]: We should be using those:

```
http://cloud.centos.org/centos/6/images/CentOS-6-x86_64-GenericCloud.qcow2
http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2
http://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.1.1911-20200113.3.x86_64.qcow2
http://download.opensuse.org/distribution/leap/15.0/jeos/openSUSE-Leap-15.0-JeOS.x86_64-15.0.1-OpenStack-Cloud-Current.qcow2
http://download.opensuse.org/distribution/leap/15.1/jeos/openSUSE-Leap-15.1-JeOS.x86_64-15.1.0-OpenStack-Cloud-Current.qcow2
http://download.opensuse.org/distribution/leap/15.2/appliances/openSUSE-Leap-15.2-JeOS.x86_64-OpenStack-Cloud.qcow2
http://download.suse.de/install/SLE-15-JeOS-GM/SLES15-JeOS.x86_64-15.0-OpenStack-Cloud-GM.qcow2
http://download.suse.de/install/SLE-15-SP1-JeOS-QU2/SLES15-SP1-JeOS.x86_64-15.1-OpenStack-Cloud-QU2.qcow2
http://download.suse.de/install/SLE-15-SP2-JeOS-GM/SLES15-SP2-JeOS.x86_64-15.2-OpenStack-Cloud-GM.qcow2
http://download.suse.de/install/SLE-15-SP3-JeOS-Alpha4/SLES15-SP3-JeOS.x86_64-15.3-OpenStack-Cloud-Alpha4.qcow2 
http://download.suse.de/install/SLE-12-SP4-JeOS-GM/SLES12-SP4-JeOS.x86_64-12.4-OpenStack-Cloud-GM.qcow2
http://download.suse.de/install/SLE-12-SP5-JeOS-GM/SLES12-SP5-JeOS.x86_64-12.5-OpenStack-Cloud-GM.qcow2
http://download.suse.de/install/SLE-15-SP3-JeOS-Beta1/SLES15-SP3-JeOS.x86_64-15.3-OpenStack-Cloud-Beta1.qcow2
```

Comment (ktsamis) [reply salt/mirror/mirror.sh.conf:8]: Yes I will copy over the contents of the file https://gitlab.suse.de/galaxy/infrastructure/-/blob/master/srv/salt/minima/mirror.sh.conf 
The 15sp3 image was the latest I could find on download.suse.de which is probably the latest milestone.

File: salt/mirror/mirror.sh.conf (hunk 2 above)
Comment (ktsamis) [reply salt/mirror/mirror.sh.conf:7]: I cppied over the contents of the file https://gitlab.suse.de/galaxy/infrastructure/-/blob/master/srv/salt/minima/mirror.sh.conf so now we should be synced. Not sure if we want to remove the older ones that are duplicate, I think they are deliberately left over to be able to use official and custom images both.

Comment (ktsamis) [reply salt/mirror/mirror.sh.conf:7]: Now we should be good to go?

**Review by Bischoff — APPROVED** [review]
LGTM, but needs testing of course, as it's easy to forget some of the needed code in sumaform.

Also take care that new variables like `SLE15SP2_MINION` will need support on test suite side (if not already done).

File: salt/mirror/mirror.sh.conf (hunk 2 above)
Comment (srbarrios) [reply salt/mirror/mirror.sh.conf:7]: Actually no :trollface: 
Sorry. I checked and we apparently have also this internal config not synced with sumaform :facepalm: 

So check this line:

https://github.com/uyuni-project/sumaform/blob/master/backend_modules/libvirt/base/main.tf#L20

You will see it differs from:
`http://download.suse.de/install/SLE-15-SP3-JeOS-TEST/SLES15-SP3-JeOS.x86_64-15.3-OpenStack-Cloud-Build20.212.qcow2`

Comment (ktsamis) [reply salt/mirror/mirror.sh.conf:7]: I updated the link there

**Review by srbarrios — APPROVED** [review]

**Pull request merged**